### PR TITLE
[Quant][X86] add an op to compute uint8 batch norm 2d

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #152811
* #152411

**Summary**
This PR adds a new op, `onednn.qbatch_norm2d`, which accepts uint8 inputs on CPU device (instead of QuantizedCPU).
The new ops are implemented with AVX512 instructions and it provides similar performance as its counterpart for QuantizedCPU device `quantized.batch_norm2d`.
The new op supports output dtypes other than uint8 (fp32, fp16 and bf16 are supported).

**Test plan**
```
pytest test/quantization/core/test_quantized_op.py -k test_int8_batch_norm_onednn
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168

### CODE REVIEW

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 1]
```diff
@@ -2521,6 +2521,182 @@ void q_batch_norm_kernel(
...
+  float* alpha_ptr = a.data_ptr<float>();
+  float* beta_ptr = b.data_ptr<float>();
+  if (output.scalar_type() == c10::ScalarType::Float) {
+    auto out_ptr = output.data_ptr<float>();
+    q_batch_norm_cpu_kernel_impl<float>(
+        N, C, HxW, in_zero_point, out_zero_point, in_ptr, alpha_ptr, beta_ptr, out_ptr
+    );
+  } else if (output.scalar_type() == c10::ScalarType::BFloat16) {
+    auto out_ptr = output.data_ptr<at::BFloat16>();
+    q_batch_norm_cpu_kernel_impl<at::BFloat16>(
```
Comment (leslie-fang-intel) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:2682]: Same comment as previous PR, let's merge these branches.

**Review by leslie-fang-intel — APPROVED** [review]

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp (hunk 1 above)
Comment (Xia-Weiwen) [reply aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp]: Thanks. I have updated this part.

File: aten/src/ATen/native/quantized/cpu/Normalization.cpp [hunk 2]
```diff
@@ -407,4 +489,8 @@ TORCH_LIBRARY_IMPL(quantized, QuantizedCPU, m) {
   m.impl(TORCH_SELECTIVE_NAME("quantized::batch_norm3d_relu"), TORCH_FN(q_batch_norm3d_impl<true>));
 }
 
+TORCH_LIBRARY_IMPL(onednn, CPU, m) {
+  m.impl(TORCH_SELECTIVE_NAME("onednn::qbatch_norm2d"), TORCH_FN(int8_batch_norm2d_cpu_impl));
```
Comment (jgong5) [inline aten/src/ATen/native/quantized/cpu/Normalization.cpp:493]: hmm, even though the op is put under "onednn" namespace but seems the implementation does not rely on onednn at all?

Comment (Xia-Weiwen) [reply aten/src/ATen/native/quantized/cpu/Normalization.cpp:493]: Yes, that's true. It's because all these ops used for PT2E quantization are put under the onednn namespace, and they should only be used when using the X86Inductor backend. We decide to keep that practice. Do you have any concerns? Thanks.

Comment (jgong5) [reply aten/src/ATen/native/quantized/cpu/Normalization.cpp:493]: Perhaps we can separate the "op" namespace and the quantization backend such as "onednn". For op, "onednn" means that the implementation relies on the oneDNN library and don't have to put other quantized ops under "onednn" namespace. On the other hand, the quantization support with "onednn" quant backend can leverage ops implemented with oneDNN library and native ATen. What do you think?

Comment (Xia-Weiwen) [reply aten/src/ATen/native/quantized/cpu/Normalization.cpp:493]: Thanks for the suggestion. It makes sense to me. How would you suggest we name the namespace? `x86` or still `quantized`? cc @leslie-fang-intel  @jerryzh168

Comment (Xia-Weiwen) [reply aten/src/ATen/native/quantized/cpu/Normalization.cpp:493]: Hi @jerryzh168 @leslie-fang-intel Do you think it's ok to use `onednn` namespace even if the we don't use oneDNN library for implementation? If not, do you have any suggestions on name of the namespace? Thanks.

Comment (jgong5) [reply aten/src/ATen/native/quantized/cpu/Normalization.cpp:493]: btw, it could be a long-term refactoring work, not necessarily blocking this PR landing.

Comment (jerryzh168) [reply aten/src/ATen/native/quantized/cpu/Normalization.cpp:493]: it can be `x86` I think, since it's used in x86 inductor backend

**Review by jerryzh168 — APPROVED** [review]

File: aten/src/ATen/native/quantized/cpu/Normalization.cpp (hunk 2 above)
Comment (Xia-Weiwen) [reply aten/src/ATen/native/quantized/cpu/Normalization.cpp:493]: @jgong5 @jerryzh168 Thanks for your suggestions. I will keep this PR as is and refactor this part in a future PR.

**Review by jgong5 — APPROVED** [review]

**Review by jerryzh168 — APPROVED** [review]
